server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -463,6 +464,7 @@ public void beforeIndexShardCreated(ShardRouting routing, Settings indexSettings
     }
 
     protected void putReaderContext(ReaderContext context) {
+        logger.trace("adding reader context [{}]", context.id());
```

Comment:
I don't think we want to accidentally merge this into main. The test seems to fail enough that it should be reproducible locally or with this PR if the test is unmuted.

Reply:
Hmm fair enough. Would you be worried of potential impact with this addition? 
Tbh, I haven't been lucky locally even after a huge number of runs. Will try to reproduce within this PR :)